src/wled2.ts

```diff
@@ -48,13 +47,102 @@ export = (RED: Red): void => {
     });
   });
 
+  // This function will 
+  function which(
+    incomingPayloadValue: string | number | boolean | IWledSegment[],
+    nodePayloadValue: string | number | boolean | IWledSegment[],
+    wrapper?: NumberConstructor // Add more types if used for others later.
+  ): string | number | boolean | IWledSegment[] {
+    const val = incomingPayloadValue ?? nodePayloadValue ?? undefined;
+    return wrapper && val !== undefined ? wrapper(val) : val
+  }
+
+  function getMergedPayloadAndConfig(payload: IWledNodeProperties, config: IWledNodeProperties): IWledNodeProperties {
+    const mergedProperties = {
+      name: "N/A"
+    } as IWledNodeProperties
+    mergedProperties.address = which(payload.address, config.address) as string
+    mergedProperties.brightness = which(payload.brightness, config.brightness, Number) as number

```

Comment:
And this could just be `Number(payload.brightness ?? config.brightness)`? There might be a bit of extra work required here to ensure no value winds up being undefined. Might take a bit of playing around to get it right.

Reply:
Disagree here, `Number(null ?? null) == 0` when in this case we want it to be undefined